File: 268.missing-number/268.missing-number.js
/*
 * @lc app=leetcode id=268 lang=javascript
 *
 * [268] Missing Number
 * Time Complexity: O(n)
 * Space Complexity: O(n)
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @return {number}
 */
var missingNumber = function (nums) {
	let array = new Array(nums.length + 1).fill(0);

	for (let i = 0; i < nums.length; i++) {
		array[nums[i]]++;
	}
	let result = -1;

	for (let i = 0; i < array.length; i++) {
		if (array[i] == 0) result = i;
	}

	return result;
};
// @lc code=end

File: 7-day-sprint/day1/268.missing-number.py
#
# @lc app=leetcode id=268 lang=python3
#
# [268] Missing Number
#


# @lc code=start
class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        bucket = [0] * (len(nums) + 1)
        for n in nums:
            bucket[n] += 1

        for i, n in enumerate(bucket):
            if bucket[i] == 0:
                return i
            
        return None
            

# @lc code=end
# Concept: Count Value

# input case:
# [0] or [1]
# [0,1] [0,2]